Library/Homebrew/api.rb

```diff
@@ -116,7 +157,18 @@ def self.fetch_homebrew_cask_source(name, git_head: nil)
         end
       end
 
-      cache[raw_endpoint] = output.stdout
+      cask_source = output.stdout
+      actual_sha256 = Digest::SHA256.hexdigest(cask_source)
+      if sha256 && actual_sha256 != sha256
+        raise <<~EOS
```

Comment:
Might want a more specific exception here and what the actionable item is for the user.

Reply:
Due to the lack of any caching, it's probably just a `brew update` and try again.

Reply:
If it's raw GitHub (hopefully almost always): I guess they're victim to a MITM? Or a corrupt download maybe. Not really much to say except "try again" maybe.

If it's formulae.brew.sh: race condition issue probably which `brew update` covers.

Reply:
> I agree this should probably use ChecksumMismatchError, though that currently references a cache which we don't have for casks yet.

Yeh, could tweak the error message/class to make the cache bit configurable?

> If it's raw GitHub (hopefully almost always): I guess they're victim to a MITM? Or a corrupt download maybe. Not really much to say except "try again" maybe.

Yeh, I think that's fine, should say "try again" or above comment version explicitly.

Reply:
> Yeh, could tweak the error message/class to make the cache bit configurable?

This will use a cache soon so will just wait a couple days for that.

Will modify the current message with "try again" though.